### PR TITLE
Ci/setup base worflows

### PR DESCRIPTION
**Pull request by hleonardtek**
The purpose of this PR is to make github-exporter compliant to OKP4 standards.

- improve doc
- set up complete CI

### CODE REVIEW

**Comment by ccamel** [comment]
@amimart @ad2ien  @hleonardtek FYI, I configured in this repository the branch protection rule for `main`, in the same way as we do for all our other opensource projects.

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+linters-settings:
+  cyclop:
+    max-complexity: 20
+    skip-tests: true
+  # funlen:
+  #   statements: 65
+  # godot:
+  #   scope: declarations # comments to be checked: `declarations` (default), `toplevel`, or `all`
+  #   exclude:
+  #     - "this line is used by starport scaffolding .*"
```
Comment (ccamel) [inline .golangci.yml]: [Please, don't commit commented out code](https://kentcdodds.com/blog/please-dont-commit-commented-out-code).

To be applied everwhere relevent in this PR. :wink:

File: README.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
 # xrstf's GitHub Exporter for Prometheus
 
-This exporter exposes Prometheus metrics for a list of pre-configured GitHub repositories.
+[![version](https://img.shields.io/github/v/release/okp4/github-exporter?style=for-the-badge&logo=github)](https://github.com/okp4/github-exporter/releases)
+[![lint](https://img.shields.io/github/workflow/status/okp4/github-exporter/Lint?label=lint&style=for-the-badge&logo=github)](https://github.com/okp4/github-exporter/actions/workflows/lint.yml)
+[![build](https://img.shields.io/github/workflow/status/okp4/github-exporter/Build?label=build&style=for-the-badge&logo=github)](https://github.com/okp4/github-exporter/actions/workflows/build.yml)
+[![conventional commits](https://img.shields.io/badge/Conventional%20Commits-1.0.0-yellow.svg?style=for-the-badge&logo=conventionalcommits)](https://conventionalcommits.org)
+[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg?style=for-the-badge)](https://github.com/okp4/.github/blob/main/CODE_OF_CONDUCT.md)
+[![License](https://img.shields.io/badge/License-BSD_3--Clause-blue.svg?style=for-the-badge)](https://opensource.org/licenses/BSD-3-Clause)
```
Comment (ccamel) [inline README.md:10]: Not correct. The license is MIT, as stated by the [LICENSE](https://github.com/okp4/github-exporter/blob/main/LICENSE) file. Moreover, this license should be amended to (1) state the copyright to the source and (2) express the terms for use of the software.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-docker:
+    runs-on: ubuntu-20.04
```
Comment (ccamel) [inline .github/workflows/build.yml]: Update running images to latest: `ubuntu-22.04`

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  build-go:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v3.0.0
+        with:
+          go-version: "^1.14"
```
Comment (ccamel) [inline .github/workflows/build.yml]: Please update go version to latest one: `1.19`.

**Review by amimart — APPROVED** [review]
Very nice thank you ! 😎

**Review by ccamel — CHANGES_REQUESTED** [review]

**Review by ad2ien — APPROVED** [review]
@hleonardtek, I added a few commits to add some consistency. 
Ok to activate some disabled linters in the future 😜

File: .github/workflows/build.yml (hunk 3 above)
Comment (ad2ien) [reply .github/workflows/build.yml]: done

File: README.md (hunk 2 above)
Comment (ad2ien) [reply README.md:10]: I mentioned the previous license and the original project at the beginning of the readme, and set the licence to BSD.

**Comment by hleonardtek** [comment]
Perfect, thank you :blush:  :+1:

File: .github/workflows/build.yml (hunk 4 above)
Comment (ad2ien) [reply .github/workflows/build.yml]: yes, done!

**Review by ad2ien — APPROVED** [review]
:clap:

**Comment by ad2ien** [comment]
@ccamel thanks we took your remarks into account. Is it ok to merge?

**Review by ccamel — APPROVED** [review]
:+1:

**Pull request merged**